src/main/java/net/neoforged/neoforge/client/model/data/ModelDataManager.java

```diff
@@ -32,49 +33,125 @@
  */
 @ApiStatus.Internal
 @EventBusSubscriber(modid = "neoforge", bus = Bus.FORGE, value = Dist.CLIENT)
-public class ModelDataManager {
-    private final Level level;
-    private final Map<ChunkPos, Set<BlockPos>> needModelDataRefresh = new ConcurrentHashMap<>();
-    private final Map<ChunkPos, Map<BlockPos, ModelData>> modelDataCache = new ConcurrentHashMap<>();
+public abstract sealed class ModelDataManager permits ModelDataManager.Active, ModelDataManager.Snapshot {
+    ModelDataManager() {}
 
-    public ModelDataManager(Level level) {
-        this.level = level;
-    }
+    public abstract void requestRefresh(BlockEntity blockEntity);
```

Comment:
How about making this a public method on `Active` since it's useless to call on a snapshot?